frontend/views/components/ContributionItem.vue

```diff
@@ -0,0 +1,152 @@
+<template lang='pug'>
+.c-contribution-item
+  i(:class='iconClass')
+
+  template(v-if='type==="MONETARY" && hasWhoElse')
+    transition(name='replacelist')
+      div(
+        v-if='isVisible'
+        key='visible'
+      )
+        .c-contribution-list(v-html='listOfName')
+
+        i18n.is-unstyled.is-link-inherit.link(
+          tag='button'
+          type='button'
+          @click='isVisible = !isVisible'
+        ) Hide
+
+      div(
+        v-else
+        key='hidden'
+        @click='isVisible = !isVisible'
+        v-html='contribution'
+      )
+
+  span.has-text-bold(
+    v-else-if='type==="NON_MONETARY" && action === "GIVING"'
+  ) {{what}}
+
+  .c-contribution-list(
+    v-else=''
+    v-html='contribution'
+  )
+</template>
+
+<script>
+import L from '@view-utils/translations.js'
+
+export default {
+  name: 'ContributionItem',
+  props: {
+    who: [String, Array],
+    what: String,
+    action: {
+      type: String,
+      validator: function (value) {
+        // The value must match one of these strings
+        return ['RECEIVING', 'GIVING'].indexOf(value) !== -1 || null
+      },
+      default: 'RECEIVING'
+    },
+    type: {
+      type: String,
+      validator: function (value) {
+        // The value must match one of these strings
+        return ['NON_MONETARY', 'MONETARY'].indexOf(value) !== -1
+      }
+    }
+  },
+  data () {
+    return {
+      isVisible: false
+    }
+  },
+  computed: {
+    contribution () {
+      if (this.hasWhoElse) {
+        const html = {
+          0: `<span class="has-text-bold">${this.what}</span>`,
+          1: `<button class="is-unstyled is-link-inherit link">${this.notFirstWho.length}`,
+          2: '</button>'
+        }
+        if (this.action === 'RECEIVING' && this.type === 'MONETARY') {
```

Comment:
Here instead of an `if` you can use a map:

```js
        switch (this.type) {
          case 'NON_MONETARY_RECIEVING':
            return L('{service} from {num} members{_button}', html)
          case 'MONETARY_GIVING':
            return L('{amount} to {num} members{_button}', html)

            // etc...
        }
```

Reply:
I don't see the point of replacing a boolean by a switch case. Maybe you could explain me why is it better in this specific case?

Reply:
OK, so having an `action` property is somewhat useful for distinguishing between the different kinds of icons in `iconClass`. We could use a switch statement there also (instead of a map), but for now we an keep the logic here and iterate later.